pkg/cgroups/cgroups.go

```diff
@@ -52,9 +53,12 @@ func CheckOrMountCgrpFS(mapRoot string) {
 		if mapRoot == "" {
 			mapRoot = cgroupRoot
 		}
-		err := cgrpCheckOrMountLocation(mapRoot)
-		// Failed cgroup2 mount is not a fatal error, sockmap will be disabled however
-		if err == nil {
+
+		if err := cgrpCheckOrMountLocation(mapRoot); err != nil {
+			if option.Config.EnableHostReachableServices {
```

Comment:
Actually, given @brb 's most recent comment:

https://github.com/cilium/cilium/pull/15999#discussion_r663119101

Maybe it's better to just always warn. We _could_ go and list all of the different features that require cgroups here and check each option to decide whether to warn or not, but then any time that changes we need to update it. I think it'll be simpler maintenance to just warn, and if the user isn't using cgroups functionality it doesn't matter. If they are using cgroups functionality, then when we actually try to use cgroups it'll complain and there _will_ be log messages to help debug.